src/NumericalAlgorithms/Interpolation/Callbacks/ObserveSurfaceIntegrals.hpp

```diff
@@ -33,33 +38,81 @@ namespace db {
 template <typename TagsList>
 class DataBox;
 }  // namespace db
+namespace observers {
+namespace ThreadedActions {
+struct WriteReductionData;
+}  // namespace ThreadedActions
+template <class Metavariables>
+struct ObserverWriter;
+}  // namespace observers
 /// \endcond
 
 namespace intrp {
 namespace callbacks {
 
+namespace detail {
+template <typename T>
+struct reduction_data_type;
+
+template <typename... Ts>
+struct reduction_data_type<tmpl::list<Ts...>> {
+  // We use ReductionData because that is what is expected by the
+  // ObserverWriter.  We do a "reduction" that involves only one core,
```

Comment:
why is this referencing cores? Do you mean PE (processing element)? Charm++ does not guarantee a specific mapping to hardware so we should avoid that assumption in documentation. Maybe the wording should be "only one processing element (often equivalent to a core),"